### PR TITLE
fix: Typescript types of some Cypress helpers

### PR DESCRIPTION
**Pull request by bnussman-akamai**
## Description 📝
- Some Typescript types for some of our Cypress helpers have incorrect typings
  - **Note** We don't typecheck our cypress tests (maybe we should in the future)
 
## Major Changes 🔄
- Change `Map`s to `Record`s where applicable 
- Removes jsdoc-like notations because I don't think they are giving us any value because we have Typescript

### Before
Screenshot 2023-09-14 at 1 43 25 PM

### After
Screenshot 2023-09-14 at 1 43 10 PM

## How to test 🧪
- Check types of `visitWithLogin` in your IDE
- Make sure types are what we actually desire (is `Record` the best fit here?)
- Verify e2e test still run as expected (they are hanging in Jenkins at the time of writing so you must test locally)
- The overall risk of this change is very low because it strictly touches types and comments

### CODE REVIEW

**Comment by jdamore-linode** [comment]
Thanks @bnussman-akamai! I'll take a closer look soon, but at a glance these all look like really sensible improvements!

File: packages/manager/cypress/support/login.ts [hunk 1]
```diff
@@ -9,7 +9,7 @@ import { apiMatcher } from 'support/util/intercepts';
 
 const overrideLocalStorage = (
   window: any,
```
Comment (abailly-akamai) [inline packages/manager/cypress/support/login.ts:11]: while you're here

```suggestion
  window: Window,
```

same with `_loginWithToken(win)` and `onBeforeLoad: (win: any) => {`

**Review by abailly-akamai — APPROVED** [review]
nice! `Record` is good as far as i can tell here

I ran a few tests locally (not the whole suite) and it looked all good. We'll find out if it affects any test directly when the suite runs again.

I agree we should run typescript on our test just the same as any other ts/tsx file in the repo, especially as more people are joining on the fun of writing CY tests. @jdamore-linode what are your thoughts on it?

**Comment by jdamore-linode** [comment]
> I agree we should run typescript on our test just the same as any other ts/tsx file in the repo, especially as more people are joining on the fun of writing CY tests. @jdamore-linode what are your thoughts on it?

@abailly-akamai @bnussman-akamai Completely agreed.

**Comment by abailly-akamai** [comment]
@jdamore-linode 👍  making a ticket for it

**Review by cpathipa — APPROVED** [review]
LGTM!, No type errors found for visitWithLogin in IDE
Inline with the usage of Record.
Below e2e tests failed, I believe these are not related to this changes. 
- account/betas.spec.ts
- loadBalancers/load-balancer-navigation.spec.ts
- longview/longview.spec.ts 
- stackscripts/smoke-community-stackscrips.spec.ts 
- vpc/vpc-navigation.spec.ts

**Pull request merged**